### ISSUE TITLE
datalad install of a git meta-dataset makes it an annex locally

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#404]
```
$> datalad install -s http://datasets.datalad.org/.git datalad
2016-04-19 10:27:53,399 [ERROR  ] stderr| Failed to get annex.uuid configuration of repository origin
| Instead, got: "core.repositoryformatversion\n0\NULcore.filemode\ntrue\NULcore.bare\nfalse\NULcore.logallrefupdates\ntrue\NULcore.sharedrepository\n2\NULreceive.denynonfastforwards\ntrue\NULreceive.denycurrentbranch\nupdateInstead\NULsubmodule.openfmri.url\n/srv/datasets.datalad.org/www/openfmri\NUL"
| This is unexpected; please check the network transport! (cmd.py:351)
1 installed item is available at
<Dataset path=/tmp/datalad>
(venv-tests)3 11086.....................................:Tue 19 Apr 2016 10:27:53 AM EDT:.
smaug:/tmp
$> cd datalad
openfmri/
(venv-tests)3 11087.....................................:Tue 19 Apr 2016 10:27:55 AM EDT:.
(git)smaug:/tmp/datalad[master]
$> cat .git/config                                            
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = http://datasets.datalad.org/.git
	fetch = +refs/heads/*:refs/remotes/origin/*
	annex-bare = false
[branch "master"]
	remote = origin
	merge = refs/heads/master
[annex]
	uuid = e34cba67-05cd-4f22-aef8-f6f461f13d0c
	version = 5
```

**Comment by hanke** [comment]
I remember this as the reason of an implementation "shortcut", but I agree this should not happen.

**Comment by yarikoptic** [comment]
I found some of crawling things all initiate annex without need... So it is a ball for me too I guess

**Comment by hanke** [comment]
The reasons seems to be that "install" unconditionally uses this call to clone repos:

````
AnnexRepo(ds.path, url=source, create=True)
````
install.py line 149

I guess we should use `GitRepo` instead and subsequently check whether something has an annex.

**Comment by hanke** [comment]
@datalad/developers 

There are decisions to be made. Currently anything we install will turn into an AnnexRepo. This makes life simple, but may not always be desired. If we stop being simple, how do we want to do it? Proposals:

1. Whenever we install something from a remote source, we use AnnexRepo() when knows_annex() returns true after we cloned it with GitRepo()

2. We use AnnexRepo() unconditionally for any new repo created from scratch.

Open question: Do we "upgrade" to AnnexRepo() when the code logic comes to an vcs.annex_add() call? Or do we honor the type of repo and revert to a git.add() in the case of a GitRepo() instance? Or do we fail?

I am somewhat leaning towards the second (take a git repo as it is), but this makes the decision logic in the code even more complex, and we also need some utility command to manually init an annex for a plain git repo (also the case if we just fail).

Please share your thoughts.

**Comment by yarikoptic** [comment]
1. sounds sensible IMHO

I think we might avoid doing any automagic git->annex conversion.  If cloned/installed repository is just a git repository and user wants to add something to annex, he does need to make it an annex repo (`git annex init`).  But the thing is that majority of our repos would be annex repos.  We could even add some checks to through a warning or an error whenever we did expect annex but got a plain git repo, but I don't think it is in `install`.

Having `annex_add` is an effect of a delayed RF of the *Repo interfaces IMHO. Eventually we should RF it to have no `git_` and `annex_` prefixes in Git/AnnexRepo.  Majority of the code should just use `repo.add` and then rely on behavior corresponding to the class.  BUT  we might still need explicit `git_add` though (no other methods come to mind atm for similar "duplicates") for whenever to force addition to git regardless of annex settings for what should get into annex (could also be done via a kwarg `force_git=None` to `AnnexRepo.add` but imho it would lead to cumbersome code at times, unless we also define it at GitRepo level and puke if False is provided).

**Comment by hanke** [comment]
Hmm, what about (2)? I thought (1) _and_ (2) would be without question.

The second part of your argument I have issues with. Somebody installs what somebody else considered a pure meta-dataset. Now they start adding files. If I get you right, this should automatically call git.add() (because that is the only thing this kind of repo can do). This will likely lead to the situation that people think they add files to a dataset like they would normally, but actually something else is happening.

If we want to make people run `git annex init`, I think the most appropriate way would be to fail and tell them.

**Comment by yarikoptic** [comment]
I am ok with 2. (especially if we document it and "unit-test" ;) )

If someone starts adding files to a pure meta-dataset, I am ok if they start adding to 'git'.  But I am not opposing 'install' failing to add any file if repository is not annex. (may be we would add "--allow-pure-git" and/or "--auto-annex-init" switches to allow "installation" into pure git repos, and/or automagic conversion)

**Comment by hanke** [comment]
See PR #421

**Comment by yarikoptic** [comment]
I guess fixed by #421
